### DISCUSSION TITLE
Question about Decode Time (DT)

### DISCUSSION THREAD

**Discussion by keanew** [redphx/better-xcloud#113] [General]
Hi, 

I'm currently testing your extension (very good work :) ) on differents browsers (smartphone, smartTV, PC, mac) and I want to understand what can upgrade or downgrade the Decode Time ? 

On a smart TV (Sony X90J) I have around 150 ms of DT, but on smartphone or computer it's very good (between 1 and 3 ms). 

Is it depending on CPU/GPU and RAM of the device ? or it's anything else ? 

I have also test on an Android Tablet running on Lineage OS and it's similar to the Smart TV. The tablet have more RAM of the smart TV.

**Comment by redphx** [comment]
Which browser are you using? If it's Kiwi then you should check this #26

**Reply by keanew** [reply]
I don't really know. I'm trying to integrate "Better xcloud" in a webview based android APK. So I don't really know what the browser the smart tv or the tablet is using. But I think this is a "chromium" browser integrate in the webview. 

No worry for usage of your code ^^. I'm just testing out this solution to verify if performances are better or not on an "native" apk. 
And the smart TV can't install Kiwi browser. So it could be a solution.

**Reply by keanew** [reply]
That's why I want to know how DT works. I want to know if it"s the device or it could be a bug like you mentioned earlier.

**Reply by redphx** [reply]
In that case it uses WebView. It's similar to Chromium but doesn't have all the features from Chromium.  
I think this is a bug of Chromium on Android. If the stream is smooth then you can just ignore this stat.

**Reply by KingNothing81** [reply]
> I'm trying to integrate "Better xcloud" in a webview based android APK.

Your approach really sounds like the perfect solution for Android TV: A dedicated app that would open the xCloud website in full screen in a webview. And that has implemented the “Better xCloud” script in the background. And that would also add support for Android TV remotes, so you wouldn’t need a mouse (or an app that simulates a mouse) to interact with the menus / settings.

I think if you were able to achieve this, me and many other Android TV users  would be very happy about this.

**Comment by ricardo404** [comment]
After I updated my Edge, the "DT" started showing these high values. Anyone else have this? Although I'm not noticing anything "heavy" something else happened.

![image](https://github.com/redphx/better-xcloud/assets/35298519/de2fe5c4-5c9e-4064-9aa6-c895e6d88fcb)

**Reply by ricardo404** [reply]
Microsoft Edge
Versão 117.0.2045.31 (Compilação oficial) (64 bits)

**Reply by redphx** [reply]
Are you using 1.16 stable or the beta version with translations?

**Reply by ricardo404** [reply]
It was and reversed, but it continues, Red. I honestly don't know what it is, but this is happening. If I move the camera in my game it increases even more. Strange. But I don't know if it's affecting my gameplay.

**Reply by redphx** [reply]
Try 1.15 and see if it still happens

**Reply by ricardo404** [reply]
![image](https://github.com/redphx/better-xcloud/assets/35298519/7bc5f801-77af-4eaf-be44-92bc7ed4d5a7)

Now the DT turns red continuously.

**Reply by ricardo404** [reply]
![image](https://github.com/redphx/better-xcloud/assets/35298519/a667fbc5-c480-4330-8570-906a62e9824c)

**Comment by ricardo404** [comment]
I have not seen it. It was after I logged into Edge and it notified me that it had updated, and it looked like this. I don't remember seeing it look like this before.

**Comment by ricardo404** [comment]
![image](https://github.com/redphx/better-xcloud/assets/35298519/d4019e12-f8c7-4e4e-bd35-c3c43b2bfada)

when it stops image transitions, it returns to low values. But normally it's not at four, but at 1. I'm going to turn off Clarity from Better, let's see.

**Reply by ricardo404** [reply]
Also, it's not!

**Comment by Spkill** [comment]
I'm having the same issue.
Both Chrome and Edge give me a 3ms DT idling and a 13ms when moving screen. Ping is less than 40ms and no matter is clarity is on or off.tried messing with setting but nothing changes. The games dont seem to be that choppy also

**Comment by STZ-Mech** [comment]
I have this exact issue with my G cloud in Kiwi browser. 25-28ms of decode time.  Is there something I can do to fix this?

**Reply by KingNothing81** [reply]
See first post:

> If it's Kiwi then you should check this https://github.com/redphx/better-xcloud/issues/26

So if you don't see stuttering or dropped frames, you can just ignore this.

**Comment by KingNothing81** [comment]
Using the System WebView based Android app also shows decode times in the same range like Kiwi: Around 20-30ms. But the stream stats also show no dropped frames. So I guess that showing decode time is just bugged on Chromium based browsers on Android.
